### PR TITLE
STORM-1614: backpressure changes and test

### PR DESCRIPTION
**Pull request by abellina**

### CODE REVIEW

File: storm-core/src/clj/org/apache/storm/daemon/worker.clj [hunk 1]
```diff
@@ -738,6 +738,7 @@
                     (run-worker-shutdown-hooks worker)
 
                     (.removeWorkerHeartbeat (:storm-cluster-state worker) storm-id assignment-id (long port))
+                    (.removeWorkerBackpressure (:storm-cluster-state worker) storm-id assignment-id (long port))
```
Comment (abellina) [inline storm-core/src/clj/org/apache/storm/daemon/worker.clj]: I tested submiting topologies and killing them and did not see issues do to this added line, but to those with more knowledge of backpressure logic please let me know if you can think of issues.

**Comment by abellina** [comment]
1.x branch PR: https://github.com/apache/storm/pull/1206

File: storm-core/src/clj/org/apache/storm/daemon/nimbus.clj [hunk 2]
```diff
@@ -1142,6 +1142,9 @@
   (blob-rm-key blob-store (ConfigUtils/masterStormConfKey id) storm-cluster-state)
   (blob-rm-key blob-store (ConfigUtils/masterStormCodeKey id) storm-cluster-state))
 
+(defn force-delete-dir [conf id]
```
Comment (zhuoliu) [inline storm-core/src/clj/org/apache/storm/daemon/nimbus.clj]: May rename this to be less general, e.g., "force-delete-topo-dist-dir"

File: storm-core/test/jvm/org/apache/storm/cluster/StormClusterStateImplTest.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+    @Test
+    public void createdZNodes() {
+        for (String path : pathlist) {
+            Mockito.verify(storage).mkdirs(path, null);
+        }
+    }
+
+    @Test
+    public void removeBackpressureDoesNotThrowTest() {
```
Comment (zhuoliu) [inline storm-core/test/jvm/org/apache/storm/cluster/StormClusterStateImplTest.java]: Thanks! Great to have a StormClusterStateImplTest.

**Comment by revans2** [comment]
+1

**Comment by ptgoetz** [comment]
+1

**Pull request merged**